applications/luci-app-acme/htdocs/luci-static/resources/view/acme.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+'require form';
+'require fs';
+'require view';
+
+return view.extend({
+    load: function () {
+        return Promise.all([
+            L.resolveDefault(fs.stat('/usr/sbin/nginx'), null),
+            L.resolveDefault(fs.stat('/usr/sbin/uhttpd'), null)
+        ]);
+    },
+
+    render: function (stats) {
+        var nginx_presence = stats[0];
+        var uhttpd_presence = stats[1];
+
+        var m = new form.Map("acme", _("ACME certificates"),
+            _("This configures ACME (Letsencrypt) automatic certificate installation. " +
+                "Simply fill out this to have the router configured with Letsencrypt-issued " +
+                "certificates for the web interface. " +
+                "Note that the domain names in the certificate must already be configured to " +
+                "point at the router's public IP address. " +
+                "Once configured, issuing certificates can take a while. " +
+                "Check the logs for progress and any errors."));
+
+        var s = m.section(form.TypedSection, "acme", _("ACME global config"));
+        s.anonymous = true
+
+        var st = s.option(form.Value, "state_dir", _("State directory"),
+            _("Where certs and other state files are kept."));
+        st.rmempty = false;
+        st.datatype = "directory";
+
+        var ae = s.option(form.Value, "account_email", _("Account email"),
+            _("Email address to associate with account key."))
+        ae.rmempty = false;
+        ae.datatype = "minlength(1)";
+
+        var d = s.option(form.Flag, "debug", _("Enable debug logging"));
+        d.rmempty = false;
+
+        var cs = m.section(form.TypedSection, "cert", _("Certificate config"))
+        cs.anonymous = false;
+        cs.addremove = true;
+
+        var e = cs.option(form.Flag, "enabled", _("Enabled"));
+        e.rmempty = false;
+
+        var us = cs.option(form.Flag, "use_staging", _("Use staging server"),
+            _("Get certificate from the Letsencrypt staging server " +
+                "(use for testing; the certificate won't be valid)."));
+        us.rmempty = false;
+
+        var kl = cs.option(form.ListValue, "keylength", _("Key size"),
+            _("Key size (and type) for the generated certificate."));
+        kl.value("2048", "RSA 2048 bits");
+        kl.value("3072", "RSA 3072 bits");
+        kl.value("4096", "RSA 4096 bits");
+        kl.value("ec-256", "ECC 256 bits");
+        kl.value("ec-384", "ECC 384 bits");
+        kl.default = "2048";
+        kl.rmempty = false;
+
+        if (uhttpd_presence) {
```

Comment:
Also use the following check https://github.com/openwrt/luci/blob/master/applications/luci-app-samba4/htdocs/luci-static/resources/view/samba4.js#L51 to decide if we use uhttps or nginx